core/arch/arm/plat-k3/drivers/ti_sci.c

```diff
@@ -522,6 +522,91 @@ int ti_sci_set_keyrev(uint32_t keyrev,
 	return 0;
 }
 
+int ti_sci_init_rng_fwl(uint16_t fwl_id, uint16_t sec_accel_region)
+{
+	uint16_t rng_region = RNG_TI_SCI_FW_RGN_ID;
+	uint8_t owner_index = OPTEE_HOST_ID;
+	uint8_t owner_privid = 0;
+	uint16_t owner_permission_bits = 0;
+	uint32_t control = 0;
+	uint32_t permissions[FWL_MAX_PRIVID_SLOTS] = { };
+	uint32_t num_perm = 0;
+	uint64_t start_address = 0;
+	uint64_t end_address = 0;
+	int ret = 0;
+
+	/* Try to claim Security Accelerator firewall for ourselves */
+	ret = ti_sci_change_fwl_owner(fwl_id, sec_accel_region, owner_index,
+				      &owner_privid, &owner_permission_bits);
+	if (ret) {
+		/*
+		 * This is not fatal, it just means we are on an HS device
+		 * where the DMSC already owns the accelerator. On GP we need
+		 * to do additional setup for access permissions below.
+		 */
+		DMSG("Could not change Security Accelerator firewall owner");
+	} else {
+		IMSG("Fixing Security Accelerator firewall owner in GP device");
+
+		/* Get current firewall configuration */
+		ret = ti_sci_get_fwl_region(fwl_id, sec_accel_region, 1,
+					    &control, permissions,
+					    &start_address, &end_address);
+		if (ret) {
+			EMSG("Could not get firewall region information");
+			return TEE_ERROR_GENERIC;
+		}
+
+		/* Modify firewall to allow all others access*/
+		control = FW_BACKGROUND_REGION | FW_ENABLE_REGION;
+		permissions[0] = (FW_WILDCARD_PRIVID << 16) | FW_NON_SECURE;
+		ret = ti_sci_set_fwl_region(fwl_id, sec_accel_region, 1,
+					    control, permissions,
+					    0x0, UINT32_MAX);
+		if (ret) {
+			EMSG("Could not set firewall region information");
+			return TEE_ERROR_GENERIC;
+		}
+	}
+	/* Claim the TRNG firewall for ourselves */
+	ret = ti_sci_change_fwl_owner(fwl_id, rng_region, owner_index,
+				      &owner_privid, &owner_permission_bits);
+	if (ret) {
+		EMSG("Could not change TRNG firewall owner");
+		return TEE_ERROR_GENERIC;
+	}
```

Comment:
This is actually unnecessary for AM62L. The eip76 TRNG block is outside the DTHEv2 block and has a separate firewall ID. So the code acquiring security accelerator region is essentially doing nothing in HS devices and is configuring 2 regions in the TRNG firewall in GP devices.